### PR TITLE
Add ingress-nginx debian image generation

### PR DESCRIPTION
**Pull request by rikatz**
We are doing an attempt to migrate to Debian as stabilize some weird coredumps errors.

So we need to regenerate all images, from the base one to the final one.

### CODE REVIEW

File: config/jobs/image-pushing/k8s-staging-ingress-nginx.yaml [hunk 1]
```diff
@@ -44,6 +44,48 @@ postsubmits:
...
 
+    - name: post-ingress-nginx-build-nginx-image-debian
+      cluster: k8s-infra-prow-build-trusted
+      annotations:
+        testgrid-dashboards: sig-network-ingress-nginx, wg-k8s-infra-gcb
+      decorate: true
+      run_if_changed: 'images/nginx-debian/.*'
+      branches:
+        - ^main$
+        - ^dev-v1$
```
Comment (tao12345666333) [inline config/jobs/image-pushing/k8s-staging-ingress-nginx.yaml:55]: Should we remove this branch?

File: config/jobs/image-pushing/k8s-staging-ingress-nginx.yaml [hunk 2]
```diff
@@ -44,6 +44,48 @@ postsubmits:
...
+
+    - name: post-ingress-nginx-build-modsecurity
+      cluster: k8s-infra-prow-build-trusted
+      annotations:
+        testgrid-dashboards: sig-network-ingress-nginx, wg-k8s-infra-gcb
+      decorate: true
+      run_if_changed: 'images/modsecurity/.*'
+      branches:
+        - ^main$
+        - ^dev-v1$
```
Comment (tao12345666333) [inline config/jobs/image-pushing/k8s-staging-ingress-nginx.yaml:76]: Ditto

File: config/jobs/image-pushing/k8s-staging-ingress-nginx.yaml [hunk 3]
```diff
@@ -65,6 +107,27 @@ postsubmits:
...
 
+    - name: post-ingress-nginx-build-e2e-image-debian
+      cluster: k8s-infra-prow-build-trusted
+      annotations:
+        testgrid-dashboards: sig-network-ingress-nginx, wg-k8s-infra-gcb
+      decorate: true
+      run_if_changed: 'images/test-runner-debian/.*'
+      branches:
+        - ^main$
+        - ^dev-v1$
```
Comment (tao12345666333) [inline config/jobs/image-pushing/k8s-staging-ingress-nginx.yaml:118]: Ditto

**Review by tao12345666333 — COMMENTED** [review]
LGTM 

Leave a comment

**Comment by rikatz** [comment]
@tao12345666333 thanks for the review.

I've changed from dev-v1 to legacy on the cases that matters (old images that we still support) and in Debian case kept only dev-v1 as this is going to be somethng new :)

**Pull request merged**